pallets/storage-provider/src/lib.rs

```diff
@@ -293,10 +298,10 @@ pub mod pallet {
             recoveries: BoundedVec<RecoveryDeclaration, ConstU32<DECLARATIONS_MAX>>,
         },
         /// Emitted when an SP doesn't submit Windowed PoSt in time and PoSt hook marks partitions as faulty
-        PartitionFaulty {
+        PartitionsFaulty {
             owner: T::AccountId,
-            partition: PartitionNumber,
-            sectors: BoundedBTreeSet<SectorNumber, ConstU32<MAX_SECTORS>>,
+            // No need to be bounded as we are creating the sets ourselves
+            faulty_partitions: BTreeMap<PartitionNumber, BTreeSet<SectorNumber>>,
```

Comment:
I don't get this comment + the bound has important information. Can you elaborate?

Reply:
The data we are returning is retrieved from data that is bounded. I didn't see a reason to have an upper limit on these and enforce more error handling because of the bound.

Reply:
I get the point but the bound also conveys information to the user, even if little since the bound is "all sectors".

I found a counter example to your suggestion from the official SDK:
https://github.com/paritytech/polkadot-sdk/blob/8216235b480630c5a69636b172f5146711c88c9a/substrate/frame/uniques/src/lib.rs#L352-L357

In general, Polkadot states that best practice is to have the bound so I'm not quite sure if we should keep it or not.

Reply:
In this case, I would also probably use BoundedVec. It's a substrate way. But If I am being honest, I am not sure exactly what are the benefits for using the bounded collections in events. Except conveying information about the max size.

Reply:
If polka has it bounded, let's make it bounded as well. Let's be consistent (I guess the tradeoff is the code with this not-funny errors to be handled when inserting into bounded collection)